test/integration-tests/tests/reality.test.ts

```diff
@@ -0,0 +1,110 @@
+const requestRealityIdHeader = (realityId: any) => ({ 'reality-id': realityId });
+const realityIdHeaderWithIncludeLinkedOper = (realityId: number) => ({
+    'reality-id': realityId,
+    'include-linked-oper': true,
+});
+
+export const firstRealityTitles: string[] = ['first', 'second', 'third', 'fourth', 'fifth'];
+const secondRealityTitles: string[] = ['1st', '2nd'];
+
+const prepareDb = async () => {
+    const author = await AuthorModelPerReality({ headers: dbRealityIdHeader(0) }).create({
+        id: 0,
+        firstName: 'Foo',
+        lastName: 'Bar',
+    });
+
+    await BookModelPerReality({ headers: dbRealityIdHeader(1) }).create(
+        await generateBookArrayWithIds(firstRealityTitles),
+    );
+
+    await BookModelPerReality({ headers: dbRealityIdHeader(2) }).create(
+        generateBookArrayWithIds(secondRealityTitles),
+    );
+
+    await BookModelPerReality({ headers: dbRealityIdHeader(3) }).create({
+        id: 0,
+        title: 'Shadow Realm',
+        author,
+    });
+};
+
+const generateBookArrayWithIds = (titleArray: string[]) => {
+    const books = [];
+    for (let i = 0; i < titleArray.length; i++) {
+        books.push({ testId: i, title: titleArray[i], dataVersion: i + 1 });
+    }
+    return books;
+};
+
+beforeEach(async () => {
+    await init();
+    await prepareDb();
+});
+
+afterEach(() => {
+    return finish();
+});
+
+describe('reality tests', () => {
+    test('fetch entities from specific reality', async () => {
+        const queryBook = `query{books{realityId}}`;
+        const realityId: number = 1;
+        const response: any = await graphQLRequest(queryBook, requestRealityIdHeader(realityId));
+        const responseRealities = [];
+        for (const book of response.books) {
+            responseRealities.push(book.realityId);
+        }
+        const uniqueResponseRealities = [...new Set(responseRealities)];
+        expect(uniqueResponseRealities.length).toBe(1);
+        expect(uniqueResponseRealities).toContain(realityId);
+    });
+
+    test('fetch entities from an empty reality', async () => {
+        const emptyResult: [] = [];
+        const queryBook = `query{books{realityId}}`;
+        const realityId: number = 999;
+        const response: any = await graphQLRequest(queryBook, requestRealityIdHeader(realityId));
+        expect(response.books).toEqual(emptyResult);
+    });
+
+    test('fetch entities without reality header', async () => {
+        const queryBook = `query{books{title}}`;
+        await expect(graphQLRequest(queryBook, undefined)).rejects.toThrow(
+            'please provide reality-id header',
+        );
+    });
+
+    test('fetch entities with invalid reality header', async () => {
+        const queryBook = `query{books{title}}`;
+        await expect(graphQLRequest(queryBook, requestRealityIdHeader('oops'))).rejects.toThrow(
+            'Unable to format headers',
+        );
+    });
+
```

Comment:
test what if someone returns referenced entities from a reality other than 0 and sends include-linked-operation - it should not return these entities!